javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,199 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object AbstractMap {
+
+  private def entryEquals[K, V](entry: Map.Entry[K, V], other: Any): Boolean = {
+    other match {
+      case other: Map.Entry[_, _] =>
+        entry.getKey == other.getKey && entry.getValue == other.getValue
```

Comment:
is it ok to you to implement it in a companion object under java.util?
Something like 

```scala
    object JavaComparableUtils {
      implcit def compareNullables(x: Any) = new {
        def ===(y: Any) = 
          if (x.asInstanceOf[AnyRef] eq null) y.asInstanceOf[AnyRef] eq null
          else x.equals(y)
      }
    }

and then modify the code with one more import and a triple = ?